bitnami/redis/templates/master/application.yaml

```diff
@@ -1,6 +1,6 @@
 {{- if or (not (eq .Values.architecture "replication")) (not .Values.sentinel.enabled) }}
 apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
-kind: StatefulSet
+kind: {{ .Values.master.kind }}
```

Comment:
Could you try running kubeval --strict with the deployment values? It seems to me that volumeClaimTemplates should not be there when using deployments. Also, I guess it should create a PVC when using Deployment type

Reply:
Please bear in mind as I stated in PR this is meant "with no persistent storage".

```sh
helm install redis-test -f values-test.yaml . --dry-run --debug > generated.yaml
# remove values section and NOTES
kubeval --strict generated.yaml

PASS - redis/templates/serviceaccount.yaml contains a valid ServiceAccount (default.redis-test)
PASS - redis/templates/secret.yaml contains a valid Secret (default.redis-test)
PASS - redis/templates/configmap.yaml contains a valid ConfigMap (default.redis-test-configuration)
PASS - redis/templates/health-configmap.yaml contains a valid ConfigMap (default.redis-test-health)
PASS - redis/templates/scripts-configmap.yaml contains a valid ConfigMap (default.redis-test-scripts)
PASS - redis/templates/headless-svc.yaml contains a valid Service (default.redis-test-headless)
PASS - redis/templates/master/service.yaml contains a valid Service (default.redis-test-master)
PASS - redis/templates/master/application.yaml contains a valid Deployment (default.redis-test-master)
```

Values used:
```yaml
architecture: standalone
auth:
  password: test1234
master:
  kind: Deployment
  disableCommands:
  - FLUSHALL
  persistence:
    enabled: false
rbac:
  create: false
replica:
  replicaCount: 0
```

Reply:
Ok, managed to update PVC for a single Deployment replica. Please let me know if this works for you.